### PR TITLE
Check if object references exist at startup

### PR DESCRIPTION
**Pull request by charlesbvll**
### Description

Currently, we only check the validity of the client/server modules after messages are exchanged. This is because sometimes import times can be long.

### Related issues/PRs

N/A

### Explanation

Instead of importing the modules on startup, we can look at the abstract syntax tree from source to check if the attribute exists on startup.

### Checklist

- [x] Implement proposed change
- [x] Update the changelog entry below
- [x] Make CI checks pass
- [x] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### Changelog entry

<skip>

### Any other comments?

N/A

### CODE REVIEW

File: src/py/flwr/client/app.py [hunk 1]
```diff
@@ -97,8 +97,18 @@ def run_client_app() -> None:
...
+        raise LoadClientAppError(error_msg) from None
+
     def _load() -> ClientApp:
-        client_app: ClientApp = load_client_app(getattr(args, "client-app"))
+        client_app = load_app(getattr(args, "client-app"), LoadClientAppError)
+
+        if not isinstance(client_app, ClientApp):
+            raise LoadClientAppError(
+                f"Attribute {getattr(args, 'client-app')} is not of type {ClientApp}",
+            ) from None
```
Comment (danieljanes) [inline src/py/flwr/client/app.py:110]: Given that we have this check after each of our three call to `load_app`, we can also move it into `load_app` by passing `ClientApp` or `ServerApp` as and additional argument.

Comment (charlesbvll) [reply src/py/flwr/client/app.py:110]: This created issues with circular imports (for type hints)

File: src/py/flwr/cli/flower_toml_test.py [hunk 2]
```diff
@@ -18,11 +18,12 @@
 import textwrap
 from typing import Any, Dict
 
+from flwr.common.object_ref import validate
```
Comment (tanertopal) [inline src/py/flwr/cli/flower_toml_test.py:21]: Tests for validate should be in `object_ref_test.py`

**Review by danieljanes — DISMISSED** [review]

File: src/py/flwr/common/object_ref.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+import importlib
+from importlib.util import find_spec
+from typing import Any, Optional, Tuple, Type
+
+OBJECT_REF_HELP_STR = """
+\nThe object reference string should have the form <module>:<attribute>. Valid
+examples include `client:app` and `project.package.module:wrapper.app`. It must
+refer to a module on the PYTHONPATH and the module needs to have the specified
+attribute.
+"""
```
Comment (danieljanes) [inline src/py/flwr/common/object_ref.py:28]:
```suggestion
OBJECT_REF_HELP_STR = """
\n\nThe object reference string should have the form <module>:<attribute>. Valid
examples include `client:app` and `project.package.module:wrapper.app`. It must
refer to a module on the PYTHONPATH and the module needs to have the specified
attribute.
"""
```

**Review by danieljanes — APPROVED** [review]

**Pull request merged**